sdk/servicebus/azure-servicebus/azure/servicebus/aio/async_message.py

```diff
@@ -193,15 +216,16 @@ async def dead_letter(self, description=None):
         To receive dead-lettered messages, use `QueueClient.get_deadletter_receiver()` or
         `SubscriptionClient.get_deadletter_receiver()`.
 
-        :param description: The reason for dead-lettering the message.
-        :type description: str
+        :param str description: The error description for dead-lettering the message.
+        :param str reason: The reason for dead-lettering the message. If `reason` is not set while `description` is
+         set, then `reason` would be set the same as `description`.
         :raises: ~azure.servicebus.common.errors.MessageAlreadySettled if the message has been settled.
         :raises: ~azure.servicebus.common.errors.MessageLockExpired if message lock has already expired.
         :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
         """
         self._is_live('dead-letter')
         details = {
-            'deadletter-reason': str(description) if description else "",
+            'deadletter-reason': str(reason) if reason else (str(description) if description else ""),
```

Comment:
Is there a limitation where description as None would be problematic?  If so, should this be something we have above?  (as opposed to allowing the caller to pass None explicitly)

There's also the slight asymmetry where we have the details object populated with a deadletter-description even if reason is the only present parameter, whereas above we don't populate the description field.    If we're keeping this for consistency/backcompat I'm fine with that, consider this a note for normalizing these behaviors for T1.

Reply:
re1: no limitation on description being None. actually we even could accept this -- `details = { 'deadletter-reason': None, 'deadletter-reason': None} `.

re2: They're two separate fields which shouldn't interfere with each other. But here is mainly for consistency/backcompat support in T1.